python/lsst/meas/deblender/plugins.py

```diff
@@ -115,6 +116,205 @@ def __str__(self):
+    data = deblender.nmf.reshape_img(data, truncate=truncate)
+
+    default_kwargs = {
+        "components": None,
+        "constraints": "S",
+    }
+    
+    for k,v in default_kwargs.items():
+        if k not in multiband_kwargs.keys():
+            multiband_kwargs[k] = v
+
+    # Update optional parameters
+    if "weights" not in multiband_kwargs:
+        if useWeights:
+            # Use the inverser variance as the weights
+            weights = 1/np.array([mimg.variance.array for mimg in maskedImages])
+
+            # Set bad pixeks weights to zero
+            maskPlane = maskedImages[0].getMask().getMaskPlaneDict()
+            badPixels = (1<<maskPlane["BAD"] |
+                         1<<maskPlane["CR"] |
+                         1<<maskPlane["NO_DATA"] |
+                         1<<maskPlane["SAT"] |
+                         1<<maskPlane["SUSPECT"])
+            fpMask = afwImage.Mask(bbox)
+            debResult.footprint.spans.setMask(fpMask, 1)
+            fpMask = ~fpMask.getArray().astype(bool)
+            mask = np.zeros(weights.shape, np.int64)
+            for fidx, mimg in enumerate(maskedImages):
+                mask[fidx] = mimg.getMask().array
+            mask = ((badPixels & mask) | fpMask).astype(bool)
+            weights[mask] = 0
+
+            # Make sure that the weights and data have the same size
+            weights = deblender.nmf.reshape_img(weights, new_shape=data.shape, truncate=truncate)
+            multiband_kwargs["weights"] = weights
+    if "psf" not in multiband_kwargs and usePsf:
+        psfs = []
+        for psf in debResult.psfs:
+            psfs.append(psf.computeKernelImage().array)
+        multiband_kwargs["psfs"] = psfs
+    # This is part of the machinery of the `deblender` and `proxmin` packages
+    # (see Moolekamp and Melchior 2017 and Melchior et al 2017)
+    if "prox_S" not in multiband_kwargs and useStrictMonotonicity:
+        kwargs = {}
+        if "l0_thresh" in multiband_kwargs:
+            kwargs["l0_thresh"] = multiband_kwargs["l0_thresh"]
+            del multiband_kwargs["l0_thresh"]
+        elif "l1_thresh" in multiband_kwargs:
+            kwargs["l1_thresh"] = multiband_kwargs["l1_thresh"]
+            del multiband_kwargs["l1_thresh"]
+        multiband_kwargs["prox_S"] = deblender.proximal.strict_monotonicity(data, peaks=peaks, **kwargs)
+    # Update the morphology before the SED in each step, since the initial SED is better constrained
+    if "update_order" not in multiband_kwargs:
+        multiband_kwargs["update_order"] = [1,0]
+
+    # When a footprint includes only non-detections
+    # (peaks in the noise too low to deblend as a source)
+    # the deblender currently fails.
+    try:
+        result = deblender.nmf.deblend(img=data, peaks=peaks, **multiband_kwargs)
+    except np.linalg.LinAlgError as e:
+        log.warn("Deblend failed catastrophically, most likely due to no signal in the footprint")
```

Comment:
That's why we're testing it, to find out when/why the deblender fails. Since we're running this on 1000's of sources I'd rather just run it once and flag all of the failed blends (and follow up on them later) than continually run the deblender and fix each failure as it comes up.